sweetest/api/v1/src/main/java/com/mysugr/sweetest/framework/base/BaseSteps.kt

```diff
@@ -0,0 +1,68 @@
+package com.mysugr.sweetest.framework.base
+
+import com.mysugr.sweetest.BASE_CLASS_DEPRECATION_MESSAGE
+import com.mysugr.sweetest.MODULE_CONFIG_DEPRECATION_MESSAGE
+import com.mysugr.sweetest.TestContext
+import com.mysugr.sweetest.framework.build.StepsBuilder
+import com.mysugr.sweetest.framework.configuration.ModuleTestingConfiguration
+import com.mysugr.sweetest.framework.coroutine.CoroutinesTestContext
+import com.mysugr.sweetest.internal.Steps
+import com.mysugr.sweetest.usecases.getCurrentTestContext
+import kotlinx.coroutines.CoroutineScope
+import kotlin.coroutines.CoroutineContext
+
+private const val TEST_CONTEXT_DEPRECATION_MESSAGE =
+    "of version 2.0.0 TestContext isn't needed and is in fact IGNORED."
+
+private const val REPLACE_WITH = "BaseSteps()"
+
+@Deprecated(BASE_CLASS_DEPRECATION_MESSAGE)
+abstract class BaseSteps
+@Deprecated("$MODULE_CONFIG_DEPRECATION_MESSAGE.", ReplaceWith("BaseSteps(testContext)"))
+private constructor(
+    private val moduleTestingConfiguration: ModuleTestingConfiguration? = null
+) : CommonBase(), Steps, CoroutineScope {
+
+    // Use this constructor!
+    @Suppress("DEPRECATION")
+    constructor() : this(null)
```

Comment:
yes, it causes compiler warnings. but i moved it to the file level, because also other stuff is causing the same warnings and deprecation on the API level is not of an issue in this class.